src/main/java/com/bobocode/petros/scaner/AnnotationDependencyClassScanner.java

```diff
@@ -1,17 +1,175 @@
+        var definition = new DependencyDefinition();
+        definition.setName(firstCharacterToLowercase(aClass.getSimpleName()));
+        definition.setQualifiedName(getQualifiedNameFromPath(aClass));
+        return definition;
+    }
+
+    private List<Class<?>> getConstructorsParam(Class<?> aClass) {
+        return Arrays.stream(getDeclaredConstructors(aClass))
+                .flatMap(a -> Arrays.stream(a.getParameterTypes()))
+                .collect(toList());
+    }
+
+    private Stream<Path> getPathStream(Path path) {
+        try {
+            return Files.walk(path);
+        } catch (IOException e) {
+            LOG.error("Package {} unsuccessfully scanned", getSimpleClassPathName(path));
+            throw new NoSuchPackageFoundException(getSimpleClassPathName(path));
+        }
+    }
+
+    private String getQualifiedNameFromPath(Class<?> aClass) {
+        return aClass.getPackage().getName() + "." + aClass.getSimpleName();
+    }
+
+    private boolean hasDefaultConstructorByPath(Path path) {
+        var hasDefaultConstructor = Arrays.stream(getDeclaredConstructors(getClassByPath(path)))
+                .anyMatch(a -> a.getParameterCount() == COUNT_OF_PARAMETER_IN_CONSTRUCTOR);
+        if (!hasDefaultConstructor) {
+            LOG.error("Class {} doesn't have default constructor", getSimpleClassPathName(path));
+            throw new DefaultConstructorNotFoundException(getSimpleClassPathName(path));
+        }
+        return true;
+    }
+
+    private Class<?> getClassWhichMarkedAsDependency(Path path) {
+        return getClassByPath(path);
+    }
+
+    private boolean hasClassMarkedAsDependency(Path path) {
+        return getClassByPath(path)
+                .isAnnotationPresent(Dependency.class);
+    }
+
+    private Class<?> getClassByPath(Path path) {
+        try {
+            return Class.forName(getSimpleClassPathName(path));
+        } catch (ClassNotFoundException e) {
+            throw new DependencyClassNotFoundException(getSimpleClassPathName(path));
+        }
+    }
+
+    private static String getSimpleClassPathName(Path path) {
+        try {
+            return path.toString().replace("/", ".")
+                    .substring(path.toString().indexOf("com"))
+                    .replace(".class", "");
+        } catch (IndexOutOfBoundsException e) {
+            throw new NoSuchPathFoundException(path.toString());
+        }
+    }
+
+    private String firstCharacterToLowercase(String str) {
+        var strName = str.substring(0, 1).toLowerCase() + str.substring(1);
+        if (strName.substring(1, 2).equals(strName.substring(1, 2).toUpperCase())) {
+            strName = str.substring(0, 2).toLowerCase() + str.substring(2);
+        }
+        return strName;
+    }
+
+    private Constructor<?>[] getDeclaredConstructors(Class<?> classByPath) {
+        return classByPath.getDeclaredConstructors();
     }
-}
+}
```

Comment:
Minor. It's better to have one empty line at the end of the file. In some projects it's even required.
Just FYI - https://stackoverflow.com/questions/4700312/empty-new-line-at-the-end-of-the-java-source-files
You may see that even github is warning about that ) 
P.S. I know I had same issue in my commits, but nobody argued with that ))))

Reply:
Done!😅